src/audio/pipeline.c

```diff
@@ -246,6 +246,25 @@ int pipeline_free(struct pipeline *p)
 	return 0;
 }
 
+static int pipeline_comp_period_frames(struct comp_dev *current)
+{
+	int samplerate;
+	int period;
+
+	period = current->pipeline->ipc_pipe.period;
+
+	if (current->output_rate)
+		samplerate = current->output_rate;
+	else
+		samplerate = current->params.rate;
+
+	/* Samplerate is in kHz and period in microseconds.
+	 * As we don't have floats use scale divider 1000000.
+	 * Also integer round up the result.
+	 */
+	return ceil_divide(samplerate * period, 1000000);
```

Comment:
#include <sof/math/numbers.h>

Reply:
I thinks it is coming from component.h... so do you want me to include it explicitly to pipeline.h?

Reply:
sorry pipeline.c...

Reply:
Yes, the best practice is to include headers explicitly, so you won't be suprised during refactor.

Reply:
ok, got it and updated.